test/test_codegen.py

```diff
@@ -0,0 +1,619 @@
+    i_di = ptk.species_index("N2")
+    x = np.zeros(ptk.num_species)
+    x[i_fu] = (alpha * phi) / (nu + alpha * phi)
+    x[i_ox] = nu * x[i_fu] / phi
+    x[i_di] = (1.0 - alpha) * x[i_ox] / alpha
+
+    # Get equilibrium composition
+    sol.TPX = t, ct.one_atm, x
+    sol.equilibrate("UV")
+    t, rho, y = sol.TDY
+    p_ct = sol.P
+
+    # Compute pressure with pyrometheus and compare to Cantera
+    p_pm = ptk.get_pressure(rho, t, y)
+    assert abs(p_ct - p_pm) / p_ct < 1.0e-12
+
+
+@pytest.mark.parametrize("mechname, fuel, stoich_ratio, dt", [("uiuc", "C2H4", 3.0,
+                                                               1e-7)])
+@pytest.mark.parametrize("usr_np", numpy_list)
+def test_get_transport_properties(mechname, fuel, stoich_ratio, dt, usr_np):
+    """This function tests that pyrometheus-generated code
+    computes transport properties (viscosity and thermal conductivity)
+    correctly by comparing against Cantera"""
+    sol = ct.Solution(f"mechs/{mechname}.cti", "gas")
+    ptk_base_cls = pyro.get_thermochem_class(sol)
+    ptk = make_jax_pyro_class(ptk_base_cls, usr_np)
+
+    # Now test mixture rules, with a reactor
+    # to get sensible mass fractions
+    init_temperature = 1200.0
+    equiv_ratio = 1.0
+    ox_di_ratio = 0.21
+
+    i_fu = sol.species_index(fuel)
+    i_ox = sol.species_index("O2")
+    i_di = sol.species_index("N2")
+
+    x = np.zeros(ptk.num_species)
+    x[i_fu] = (ox_di_ratio*equiv_ratio)/(stoich_ratio+ox_di_ratio*equiv_ratio)
+    x[i_ox] = stoich_ratio*x[i_fu]/equiv_ratio
+    x[i_di] = (1.0-ox_di_ratio)*x[i_ox]/ox_di_ratio
+
+    sol.TPX = init_temperature, ct.one_atm, x
+    reactor = ct.IdealGasConstPressureReactor(sol)
+    sim = ct.ReactorNet([reactor])
+
+    dt = 1.0e-7
+    time = 0.0
+
+    for _ in range(3000):
+        time += dt
+        sim.advance(time)
+
+        # Cantera transport
+        sol.TPY = reactor.T, ct.one_atm, reactor.Y
+
+        mu_ct = sol.viscosity
+        kappa_ct = sol.thermal_conductivity
+        diff_ct = sol.mix_diff_coeffs
+
+        # Get state from Cantera
+        temp = reactor.T
+        y = np.where(reactor.Y > 0, reactor.Y, 0)
+
+        # Pyrometheus transport
+        mu = ptk.get_mixture_viscosity(temp, y)
+        kappa = ptk.get_mixture_thermal_conductivity(temp, y)
+        diff = ptk.get_mixture_diffusivity(temp, ct.one_atm, y)
+
+        err_mu = np.abs(mu - mu_ct)
+        assert err_mu < 1.0e-13
+
+        err_kappa = np.abs(kappa - kappa_ct)
+        assert err_kappa < 1.0e-13
+
+        for i in range(sol.n_species):
+            err_diff = np.abs(diff[i] - diff_ct[i])
+            assert err_diff < 1.0e-13
+
+    return
+
+
+@pytest.mark.parametrize("mechname, fuel, stoich_ratio, dt", [("uiuc", "C2H4", 3.0,
+                                                               1e-7)])
+@pytest.mark.parametrize("usr_np", numpy_list)
+def test_get_transport_properties_temp(mechname, fuel, stoich_ratio, dt, usr_np):
```

Comment:
I split the previous test in two. Now the temperature test exercise both individual species and mixtures. The mixture test is necessary for the species mass diff. since it is also a function of the mixture, so it can not be evaluated for a single species only. 

Let me know your thoughts on this.